beam/tools/src/main/java/cz/o2/proxima/beam/tools/groovy/BeamStream.java

```diff
@@ -1411,10 +1425,18 @@ public void onTimer(
     }
   }
 
-  private static <T> DoFn<T, Void> asDoFn(Consumer<T> consumer) {
+  private <T> DoFn<T, Void> asDoFn(PipelineOptions opts, Consumer<T> consumer) {
+    if (supportsStableInput(opts)) {
```

Comment:
why is this needed? this would result in increased latency when consuming remote output

Reply:
Because otherwise the consumer will see duplicates.